thewalrus/internal_modes/fock_density_matrices.py

```diff
@@ -178,4 +184,40 @@ def density_matrix_single_mode(cov, pattern, normalize=False, LO_overlap=None, c
     for i in range(K):
         U_K[i::K, i::K] = Uswap
     _, cov = passive_transformation(np.zeros(cov.shape[0]), cov, U_K, hbar=hbar)
-    return _density_matrix_single_mode(cov, N_nums, normalize, LO_overlap, cutoff, hbar)
+
```

Comment:
Everything above this is repeated multiple times (once here and twice in pnr_statistics.py. Can we put it in its own function to avoid repetition? I would also include `project_onto_local_oscillator` and take that out of `_density_matrix_single_mode`